### PR TITLE
remove useless check and create a pod copy

### PR DESCRIPTION
**Pull request by zouyee**
**What this PR does / why we need it**:
remove useless check and create a pod copy

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/volume/expand/expand_controller.go [hunk 1]
```diff
@@ -171,12 +171,6 @@ func (expc *expandController) deletePVC(obj interface{}) {
 }
```
Comment (liggitt) [inline pkg/controller/volume/expand/expand_controller.go]: The check above misses deletion tombstone final state unknown objects and means they live in the resize map forever. @gnufied

File: pkg/controller/volume/expand/expand_controller.go [hunk 2]
```diff
@@ -171,12 +171,6 @@ func (expc *expandController) deletePVC(obj interface{}) {
...
 
 func (expc *expandController) pvcUpdate(oldObj, newObj interface{}) {
-	oldPvc, ok := oldObj.(*v1.PersistentVolumeClaim)
-
-	if oldPvc == nil || !ok {
-		return
-	}
-
```
Comment (liggitt) [inline pkg/controller/volume/expand/expand_controller.go]: I expected the old and new objects to be compared to see if they needed to be added to the resize map. @gnufied

Comment (gnufied) [reply pkg/controller/volume/expand/expand_controller.go]: It does gets check before being added to the map - https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/expand/cache/volume_resize_map.go#L96

File: pkg/controller/volume/expand/expand_controller.go (hunk 1 above)
Comment (gnufied) [reply pkg/controller/volume/expand/expand_controller.go]: I do not understand - the resizeMap gets reset to empty on each iteration of loop - https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/expand/cache/volume_resize_map.go#L128

File: pkg/controller/volume/expand/expand_controller.go (hunk 2 above)
Comment (gnufied) [reply pkg/controller/volume/expand/expand_controller.go]: Okay, we don't compare old and new pvc objects before adding to resize map and that is intentional. The reason is - we can't only rely on events for processing resize requests and there is a different loop that also polls for all PVCs in cluster and checks if pvc's requested size and actual capacity don't match. If PVC is in that state, it gets added to resize map.

This is as per design and there are checks in place to make sure everything works as expected.

File: pkg/controller/volume/expand/expand_controller.go (hunk 1 above)
Comment (liggitt) [reply pkg/controller/volume/expand/expand_controller.go]: deletePVC may be passed a cache.DeletedFinalStateUnknown object, not a v1.PersistentVolumeClaim... it needs to handle that

Comment (jsafrane) [reply pkg/controller/volume/expand/expand_controller.go]: yes, the controller should handle DeletedFinalStateUnknown, however in a separate PR (or issue).
@gnufied, would you mind fixing this?

**Pull request closed without merging**